roofit/roostats/src/AsymptoticCalculator.cxx

```diff
@@ -1036,6 +1041,8 @@ RooAbsData * AsymptoticCalculator::GenerateCountingAsimovData(RooAbsPdf & pdf, c
         pois->setNoRounding(true);
     } else if ((gaus = dynamic_cast<RooGaussian *>(&pdf)) != 0) {
         r = SetObsToExpected(*gaus, observables);
+    } else if ((mvgaus = dynamic_cast<RooMultiVarGaussian *>(&pdf)) != 0) {
+        r = SetObsToExpected(*mvgaus, observables);	
```

Comment:
Are you sure this is working?

I took a look of the implementation of [SetObsToExpected](https://github.com/root-project/root/blob/master/roofit/roostats/src/AsymptoticCalculator.cxx#L967), and it seems to me that it only tries to find one observable and one expected value parameter in the PDF. This would not work for the `RooMultiVarGaussian` that has more that one observable and expected value.

Maybe it's better to implement a dedicated 'private' overload of `SetObsToExpected` for the `RooMultiVarGaussian` that is called here? And then in that overload, it does exactly the right thing for that class.

Reply:
I haven't been able to test it yet due to the issues with binned vs. unbinned asimov dataset generation.
You're probably right that factorizing this code here makes more sense.